### ISSUE TITLE
BetterFPS causes patching to crash

### ISSUE THREAD

**Issue by LunNova** [MinimallyCorrect/TickProfiler#43]
It does something weird.

```
[17:48:24] [main/ERROR]: Error patching func_72939_s in net.minecraft.world.World with insertCodeBefore
java.lang.RuntimeException: java.io.EOFException
    at javassist.CtClassType.getClassFile2(CtClassType.java:205) ~[CtClassType.class:?]
    at javassist.CtClassType.makeFieldCache(CtClassType.java:848) ~[CtClassType.class:?]
    at javassist.CtClassType.getMembers(CtClassType.java:839) ~[CtClassType.class:?]
    at javassist.CtClassType.getDeclaredField2(CtClassType.java:973) ~[CtClassType.class:?]
    at javassist.CtClassType.getField2(CtClassType.java:927) ~[CtClassType.class:?]
    at javassist.CtClassType.getField(CtClassType.java:908) ~[CtClassType.class:?]
    at javassist.CtClass.getField(CtClass.java:798) ~[CtClass.class:?]
    at javassist.compiler.MemberResolver.lookupFieldByJvmName2(MemberResolver.java:282) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess2(TypeChecker.java:887) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:844) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:660) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:157) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.CallExpr.accept(CallExpr.java:46) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:242) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.compileBooleanExpr(CodeGen.java:236) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.atIfStmnt(CodeGen.java:384) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.atStmnt(CodeGen.java:355) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Stmnt.accept(Stmnt.java:50) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.Javac.compileStmnt(Javac.java:569) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.CtBehavior.insertBefore(CtBehavior.java:754) ~[CtBehavior.class:?]
    at javassist.CtBehavior.insertBefore(CtBehavior.java:734) ~[CtBehavior.class:?]
    at me.nallar.javapatcher.patcher.Patches.insertCodeBefore(Patches.java:931) ~[Patches.class:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:406) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:376) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$ClassPatchDescriptor.runPatches(Patcher.java:501) [Patcher$ClassPatchDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher.patch(Patcher.java:191) [Patcher.class:?]
    at me.nallar.modpatcher.ModPatcher.postSrgTransformationHook(ModPatcher.java:82) [ModPatcher.class:?]
    at me.nallar.modpatcher.ModPatcher.transform(ModPatcher.java:121) [ModPatcher.class:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass1(Native Method) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass(ClassLoader.java:760) [?:1.8.0_45]
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) [?:1.8.0_45]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.Class.getDeclaredMethods0(Native Method) [?:1.8.0_45]
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701) [?:1.8.0_45]
    at java.lang.Class.privateGetMethodRecursive(Class.java:3048) [?:1.8.0_45]
    at java.lang.Class.getMethod0(Class.java:3018) [?:1.8.0_45]
    at java.lang.Class.getMethod(Class.java:1784) [?:1.8.0_45]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:132) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:43) [Thermos49.jar:1.7.10-1614.49]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:12) [Thermos49.jar:1.7.10-1614.49]
Caused by: java.io.EOFException
    at java.io.DataInputStream.readInt(DataInputStream.java:392) ~[?:1.8.0_45]
    at javassist.bytecode.ClassFile.read(ClassFile.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.bytecode.ClassFile.<init>(ClassFile.java:124) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.CtClassType.getClassFile2(CtClassType.java:192) ~[CtClassType.class:?]
    ... 61 more
[17:48:24] [main/ERROR]: Error patching updateEntities_withProfiling in net.minecraft.world.World with replaceMethodCall
java.lang.RuntimeException: java.io.EOFException
    at javassist.CtClassType.getClassFile2(CtClassType.java:205) ~[CtClassType.class:?]
    at javassist.CtClassType.makeFieldCache(CtClassType.java:848) ~[CtClassType.class:?]
    at javassist.CtClassType.getMembers(CtClassType.java:839) ~[CtClassType.class:?]
    at javassist.CtClassType.getDeclaredField2(CtClassType.java:973) ~[CtClassType.class:?]
    at javassist.CtClassType.getField2(CtClassType.java:927) ~[CtClassType.class:?]
    at javassist.CtClassType.getField(CtClassType.java:908) ~[CtClassType.class:?]
    at javassist.CtClass.getField(CtClass.java:798) ~[CtClass.class:?]
    at javassist.compiler.MemberResolver.lookupFieldByJvmName2(MemberResolver.java:282) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess2(TypeChecker.java:887) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:844) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:660) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:157) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.CallExpr.accept(CallExpr.java:46) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:242) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.atStmnt(CodeGen.java:330) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Stmnt.accept(Stmnt.java:50) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.Javac.compileStmnt(Javac.java:569) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.expr.MethodCall.replace(MethodCall.java:235) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at me.nallar.javapatcher.patcher.Patches$6.edit(Patches.java:660) ~[Patches$6.class:?]
    at javassist.expr.ExprEditor.loopBody(ExprEditor.java:192) ~[ExprEditor.class:?]
    at javassist.expr.ExprEditor.doit(ExprEditor.java:91) ~[ExprEditor.class:?]
    at javassist.CtBehavior.instrument(CtBehavior.java:712) ~[CtBehavior.class:?]
    at me.nallar.javapatcher.patcher.Patches.replaceMethodCall(Patches.java:636) ~[Patches.class:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:406) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:376) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$ClassPatchDescriptor.runPatches(Patcher.java:501) [Patcher$ClassPatchDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher.patch(Patcher.java:191) [Patcher.class:?]
    at me.nallar.modpatcher.ModPatcher.postSrgTransformationHook(ModPatcher.java:82) [ModPatcher.class:?]
    at me.nallar.modpatcher.ModPatcher.transform(ModPatcher.java:121) [ModPatcher.class:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass1(Native Method) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass(ClassLoader.java:760) [?:1.8.0_45]
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) [?:1.8.0_45]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.Class.getDeclaredMethods0(Native Method) [?:1.8.0_45]
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701) [?:1.8.0_45]
    at java.lang.Class.privateGetMethodRecursive(Class.java:3048) [?:1.8.0_45]
    at java.lang.Class.getMethod0(Class.java:3018) [?:1.8.0_45]
    at java.lang.Class.getMethod(Class.java:1784) [?:1.8.0_45]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:132) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:43) [Thermos49.jar:1.7.10-1614.49]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:12) [Thermos49.jar:1.7.10-1614.49]
Caused by: java.io.EOFException
    at java.io.DataInputStream.readInt(DataInputStream.java:392) ~[?:1.8.0_45]
    at javassist.bytecode.ClassFile.read(ClassFile.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.bytecode.ClassFile.<init>(ClassFile.java:124) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.CtClassType.getClassFile2(CtClassType.java:192) ~[CtClassType.class:?]
    ... 70 more
[17:48:24] [main/ERROR]: Error patching updateEntities_withProfiling in net.minecraft.world.World with replaceMethodCall
java.lang.RuntimeException: java.io.EOFException
    at javassist.CtClassType.getClassFile2(CtClassType.java:205) ~[CtClassType.class:?]
    at javassist.CtClassType.makeFieldCache(CtClassType.java:848) ~[CtClassType.class:?]
    at javassist.CtClassType.getMembers(CtClassType.java:839) ~[CtClassType.class:?]
    at javassist.CtClassType.getDeclaredField2(CtClassType.java:973) ~[CtClassType.class:?]
    at javassist.CtClassType.getField2(CtClassType.java:927) ~[CtClassType.class:?]
    at javassist.CtClassType.getField(CtClassType.java:908) ~[CtClassType.class:?]
    at javassist.CtClass.getField(CtClass.java:798) ~[CtClass.class:?]
    at javassist.compiler.MemberResolver.lookupFieldByJvmName2(MemberResolver.java:282) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess2(TypeChecker.java:887) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:844) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:834) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atFieldRead(TypeChecker.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atExpr(TypeChecker.java:578) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Expr.accept(Expr.java:68) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:660) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:157) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.CallExpr.accept(CallExpr.java:46) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:242) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.CodeGen.atStmnt(CodeGen.java:330) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.ast.Stmnt.accept(Stmnt.java:50) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.compiler.Javac.compileStmnt(Javac.java:569) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.expr.MethodCall.replace(MethodCall.java:235) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at me.nallar.javapatcher.patcher.Patches$6.edit(Patches.java:660) ~[Patches$6.class:?]
    at javassist.expr.ExprEditor.loopBody(ExprEditor.java:192) ~[ExprEditor.class:?]
    at javassist.expr.ExprEditor.doit(ExprEditor.java:91) ~[ExprEditor.class:?]
    at javassist.CtBehavior.instrument(CtBehavior.java:712) ~[CtBehavior.class:?]
    at me.nallar.javapatcher.patcher.Patches.replaceMethodCall(Patches.java:636) ~[Patches.class:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:406) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$PatchMethodDescriptor.run(Patcher.java:376) [Patcher$PatchMethodDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher$ClassPatchDescriptor.runPatches(Patcher.java:501) [Patcher$ClassPatchDescriptor.class:?]
    at me.nallar.javapatcher.patcher.Patcher.patch(Patcher.java:191) [Patcher.class:?]
    at me.nallar.modpatcher.ModPatcher.postSrgTransformationHook(ModPatcher.java:82) [ModPatcher.class:?]
    at me.nallar.modpatcher.ModPatcher.transform(ModPatcher.java:121) [ModPatcher.class:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass1(Native Method) [?:1.8.0_45]
    at java.lang.ClassLoader.defineClass(ClassLoader.java:760) [?:1.8.0_45]
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) [?:1.8.0_45]
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182) [launchwrapper-1.12.jar:?]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) [?:1.8.0_45]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_45]
    at java.lang.Class.getDeclaredMethods0(Native Method) [?:1.8.0_45]
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701) [?:1.8.0_45]
    at java.lang.Class.privateGetMethodRecursive(Class.java:3048) [?:1.8.0_45]
    at java.lang.Class.getMethod0(Class.java:3018) [?:1.8.0_45]
    at java.lang.Class.getMethod(Class.java:1784) [?:1.8.0_45]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:132) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_45]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:43) [Thermos49.jar:1.7.10-1614.49]
    at cpw.mods.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:12) [Thermos49.jar:1.7.10-1614.49]
Caused by: java.io.EOFException
    at java.io.DataInputStream.readInt(DataInputStream.java:392) ~[?:1.8.0_45]
    at javassist.bytecode.ClassFile.read(ClassFile.java:777) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.bytecode.ClassFile.<init>(ClassFile.java:124) ~[TickProfiler-1.7.10.jenkins.29%20(1).jar:?]
    at javassist.CtClassType.getClassFile2(CtClassType.java:192) ~[CtClassType.class:?]
    ... 70 more
```

**Comment by LunNova** [comment]
This should be resolved in 1.8.9 releases, not that that helps here.

**Comment by LunNova** [comment]
I'm not really interested in further work on 1.7.10 for issues like this which will impact a small number of people. Most users of TickProfiler will be servers, and they can't even run that mod.
